tests/unit/test_diff.py

```diff
@@ -409,3 +409,56 @@ def test_archive_diff_json_parser(line, expected):
 
     assert item.path == PurePath(expected[0]).parts
     assert item.data == DiffData(*expected[1:])
+
+
+def test_diff_item_copy(qapp, qtbot, mocker, borg_json_output):
+    main = qapp.main_window
+    tab = main.archiveTab
+    main.tabWidget.setCurrentIndex(3)
+
+    tab.populate_from_profile()
+    qtbot.waitUntil(lambda: tab.archiveTable.rowCount() == 2)
+
+    stdout, stderr = borg_json_output("diff_archives")
+    popen_result = mocker.MagicMock(stdout=stdout, stderr=stderr, returncode=0)
+    mocker.patch.object(vorta.borg.borg_job, 'Popen', return_value=popen_result)
+
+    compat = vorta.utils.borg_compat
+
+    def check(feature_name):
+        if feature_name == 'DIFF_JSON_LINES':
+            return False
+        return vorta.utils.BorgCompatibility.check(compat, feature_name)
+
+    mocker.patch.object(vorta.utils.borg_compat, 'check', check)
+
+    selection_model: QItemSelectionModel = tab.archiveTable.selectionModel()
+    model = tab.archiveTable.model()
+
+    flags = QItemSelectionModel.SelectionFlag.Rows
+    flags |= QItemSelectionModel.SelectionFlag.Select
+
+    selection_model.select(model.index(0, 0), flags)
+    selection_model.select(model.index(1, 0), flags)
+
+    tab.diff_action()
+
+    qtbot.waitUntil(lambda: hasattr(tab, '_resultwindow'), **pytest._wait_defaults)
+
+    # save original clipboard text to reset later.
+    original_clipboard = None
+    if qapp.clipboard().mimeData().hasText():
+        original_clipboard = QMimeData()
+        original_clipboard.setText(qapp.clipboard().mimeData().text())
```

Comment:
I don't _love_ this way of saving the previous clipboard and then reverting it back at the end. Considered using a CM but not sure if it would have the desired result?

Reply:
Mocked the clipboard, now the users clipboard isn't altered during testing. Marking this as resolved.